DSharpPlus/Logging/DefaultLogger.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+
+namespace DSharpPlus
+{
+    public class DefaultLogger : ILogger<BaseDiscordClient>
+    {
+        private static readonly object _lock = new object();
+
+        private LogLevel MinimumLevel { get; }
+        private string TimestampFormat { get; }
+
+        internal DefaultLogger(BaseDiscordClient client)
+            : this(client.Configuration.MinimumLogLevel, client.Configuration.LogTimestampFormat)
+        { }
+
+        internal DefaultLogger(LogLevel minLevel = LogLevel.Information, string timestampFormat = "yyyy-MM-dd HH:mm:ss zzz")
+        {
+            this.MinimumLevel = minLevel;
+            this.TimestampFormat = timestampFormat;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!this.IsEnabled(logLevel))
+                return;
+
+            lock (_lock)
+            {
+                Console.Write($"[{DateTimeOffset.Now.ToString(this.TimestampFormat)}] [{eventId.Id,-4}/{eventId.Name,-12}] ");
+
+                switch (logLevel)
+                {
+                    case LogLevel.Trace:
+                        Console.ForegroundColor = ConsoleColor.Gray;
+                        break;
+
+                    case LogLevel.Debug:
+                        Console.ForegroundColor = ConsoleColor.Magenta;
```

Comment:
This is a tiny one, but I found that setting Debug to `ConsoleColor.Green` and Trace to `ConsoleColor.Magenta` looks nice.

Reply:
Trace is not really supposed to stand out. It's meant to be the spammy level which might contain sensitive data. In general, I was considering making whole trace messages have the muted gray colour.

Reply:
That's fair. In that case I still think that Debug should be green because it is easier to see, but that's just me.

Speaking of trace and it containing sensitive data, we should probably include what was used in https://github.com/DSharpPlus/DSharpPlus/pull/464 with exposing the raw payloads, and indicate whether that data is being received or sent.

Reply:
Ultimately, that might happen. I shifted several messages to Trace level.

Reply:
Regarding colours, that's open to debate. In the end I believe it's very much a personal preference.

Reply:
Yeah, that would be cool if we were able to specify them somehow for the default logger.

Reply:
I think that's a tad excessive. It's meant to be a basic logger.

Reply:
That's true.

Reply:
This discussion would be more productive if more than two people chimed in.

Reply:
I like pink. All the colors should be pink.

In all seriousness - the more often a type of message shows up, the more reason it has to not be an intense color. Trace is that taken to the extreme. Under the same pretense, use a dark color variant for the Debug level. The Information color is fine, but may fit better as DarkCyan instead.

Reply:
Fair enough. Warning, error, and critical colours are chosen to stand out - they should, regardless of volume. I will switch info and debug to dark variants.